crates/iceberg/src/spec/values.rs

```diff
@@ -0,0 +1,964 @@
+            NaiveTime::from_num_seconds_from_midnight_opt(0, 0).unwrap(),
+        )
+        .num_microseconds()
+        .unwrap()
+    }
+
+    pub(crate) fn microseconds_to_time(micros: i64) -> NaiveTime {
+        let (secs, rem) = (micros / 1_000_000, micros % 1_000_000);
+
+        NaiveTime::from_num_seconds_from_midnight_opt(secs as u32, rem as u32 * 1_000).unwrap()
+    }
+}
+
+mod timestamp {
+    use chrono::NaiveDateTime;
+
+    pub(crate) fn datetime_to_microseconds(time: &NaiveDateTime) -> i64 {
+        time.timestamp_micros()
+    }
+
+    pub(crate) fn microseconds_to_datetime(micros: i64) -> NaiveDateTime {
+        let (secs, rem) = (micros / 1_000_000, micros % 1_000_000);
+
+        // This shouldn't fail until the year 262000
+        NaiveDateTime::from_timestamp_opt(secs, rem as u32 * 1_000).unwrap()
+    }
+}
+
+mod timestamptz {
+    use chrono::{DateTime, NaiveDateTime, Utc};
+
+    pub(crate) fn datetimetz_to_microseconds(time: &DateTime<Utc>) -> i64 {
+        time.timestamp_micros()
+    }
+
+    pub(crate) fn microseconds_to_datetimetz(micros: i64) -> DateTime<Utc> {
+        let (secs, rem) = (micros / 1_000_000, micros % 1_000_000);
+
+        DateTime::<Utc>::from_utc(
+            // This shouldn't fail until the year 262000
+            NaiveDateTime::from_timestamp_opt(secs, rem as u32 * 1_000).unwrap(),
+            Utc,
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::spec::datatypes::{ListType, MapType, NestedField, StructType};
+
+    use super::*;
+
+    fn check_json_serde(json: &str, expected_literal: Literal, expected_type: &Type) {
+        let raw_json_value = serde_json::from_str::<JsonValue>(json).unwrap();
+        let desered_literal =
+            Literal::try_from_json(raw_json_value.clone(), expected_type).unwrap();
+        assert_eq!(desered_literal, Some(expected_literal.clone()));
+
+        let expected_json_value: JsonValue = (&expected_literal).into();
+        let sered_json = serde_json::to_string(&expected_json_value).unwrap();
+        let parsed_json_value = serde_json::from_str::<JsonValue>(&sered_json).unwrap();
+
+        assert_eq!(parsed_json_value, raw_json_value);
+    }
+
+    fn check_avro_bytes_serde(input: Vec<u8>, expected_literal: Literal, expected_type: &Type) {
+        let raw_schema = r#""bytes""#;
+        let schema = apache_avro::Schema::parse_str(raw_schema).unwrap();
+
+        let bytes = ByteBuf::from(input);
+        let literal = Literal::try_from_bytes(&bytes, expected_type).unwrap();
+        assert_eq!(literal, expected_literal);
+
+        let mut writer = apache_avro::Writer::new(&schema, Vec::new());
+        writer.append_ser(bytes).unwrap();
+        let encoded = writer.into_inner().unwrap();
+        let reader = apache_avro::Reader::new(&*encoded).unwrap();
+
+        for record in reader {
+            let result = apache_avro::from_value::<ByteBuf>(&record.unwrap()).unwrap();
```

Comment:
Sorry, I can't figure it out what is the difference between result and bytes? 🤔

Reply:
Good catch! The `literal` is supposed to be written to the Avro Writer instead of the `bytes`. Like so:
```rust
writer.append_ser(Into::<ByteBuf>::into(literal)).unwrap();
```
I will create a PR to fix this.

Reply:
I created a [PR](https://github.com/apache/iceberg-rust/pull/80) to fix it.